react/src/layout/report.tsx

```diff
@@ -351,6 +360,11 @@ export class SimulationStartLayout extends Layout<SimulationStartConfig, {}> {
         let endSimulationButton = <Button variant="primary" onClick={endSimulation}>End Simulation</Button>;
```

Comment:
Is "primary" a concept we defined? If so, what are the choices? What happens if there's a typo?

The text `End Simulation` should come from somewhere. Sometimes it is `End Simulation` and other times it is something else, e.g. in Activait. Embedding strings throughout the app is going to be problematic, because they get repeated.

Reply:
In the current schema we have "strings" which get used to create these names. Ex for raydata https://github.com/radiasoft/sirepo/blob/master/sirepo/package_data/static/json/raydata-schema.json#L68

Reply:
Primary is a concept defined by Bootstrap. While I agree that we should not have too many strings floating around in our app there is no benefit in duplicating these definitions inside the schema. If we want to make this a parameter for the layout within the schema that is fine. The strings area of the schema should be used for things that potentially change but primary is just the style for the button. This is more suited as a global for the React app if anything. I believe primary is actually the default here though, so for now I would personally lean towards leaving it there. These are reusable components and their definitions will not be duplicated elsewhere so I think consistency is the most important thing here. All simulation start layouts will have the same kind of button and primary happens to be the name of it.

Reply:
The signal to noise ratio of this line is high. There's a lot of syntax and the words button and simulation are repeated three times:

`let endSimulationButton = <Button variant="primary" onClick={endSimulation}>End Simulation</Button>;`

Our minds have a harder time with differences (what's the meaning here) when there is a lot of repetition.

The danger I see of `"primary"` is that it is interpreted at run-time(?), and if it is CSS, it would not fail fast if there was a typo. Is that true?

> All simulation start layouts will have the same kind of button and primary happens to be the name of it.

This suggests a default override. There are buttons with danger and primary, and the default is primary. The thing that defines layouts should default to primary (written once in a "parent" of some sort) and the override would happen when we needed it.

Reply:
> The danger I see of "primary" is that it is interpreted at run-time(?), and if it is CSS, it would not fail fast if there was a typo. Is that true?

Since it is typescript it would raise an error while compiling. The definition for `variant` probably looks like `let type ButtonVariant = "primary" | "secondary" | "danger" //... etc`. There are checks we can use to prevent typos for strings like this. Adding them to some constant like `BUTTON_PRIMARY_NAME = "primary"` would feel redundant and if your intended approach is something like an enum then the type definition above would be an example of how enums are created in typescript.